src/datasets/load.py

```diff
@@ -619,11 +687,40 @@ def __init__(
 
         self.path = path
         self.name = Path(path).stem
+        self.config_name = config_name if config_name else "default"
```

Comment:
We can't say that the config name should "default" at this point instead of `None`

For example `squad` has one config name "plain_text" defined in the readme (and also in the builder class)